core/src/main/scala/org/apache/spark/TaskEndReason.scala

```diff
@@ -168,6 +168,23 @@ case class ExecutorLostFailure(execId: String) extends TaskFailedReason {
 
 /**
  * :: DeveloperApi ::
+ * The task failed because the executor that it was running on was prematurely terminated. The
+ * executor is forcibly exited but the exit should be considered as part of normal cluster
+ * behavior.
+ */
+@DeveloperApi
+case class ExecutorForTaskExited(
```

Comment:
This name is kind of generic. I think it's OK to just call this `ExecutorNormalExit` or something.